### PR TITLE
[CI/Docker] Upgrade `xgboost-ray` to latest release

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: amogkam <amogkamsetty@yahoo.com>

Upgrade `xgboost-ray` to latest release in ML Docker image to address release blocker.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by cadedaniel** [comment]
Will this need product validation?

**Comment by amogkam** [comment]
I think @Yard1 has already tested this out

**Comment by zhe-thoughts** [comment]
For precaution: @Yard1 could you confirm the product validation? After that, I would approve for merging

**Comment by Yard1** [comment]
I'll do it first thing tomorrow

**Comment by Yard1** [comment]
@amogkam can we also update lightgbm-ray to latest? It will conflict otherwise

File: python/requirements/ml/requirements_upstream.txt
Comment (amogkam) [inline python/requirements/ml/requirements_upstream.txt:8]:
```suggestion
lightgbm_ray==0.1.8
```

**Comment by Yard1** [comment]
Confirmed it works on product (use the nightly image, manually install latest xgboost-ray with `pip install --user`, run an example script - no issues).

File: python/requirements/ml/requirements_upstream.txt
Comment (amogkam) [inline python/requirements/ml/requirements_upstream.txt:7]:
```suggestion
xgboost_ray==0.1.15
```

**Review by Yard1 — APPROVED** [review]

**Comment by cadedaniel** [comment]
> Confirmed it works on product (use the nightly image, manually install latest xgboost-ray with `pip install --user`, run an example script - no issues).

Thanks @Yard1. Ok, I want to understand the risk here before merging -- what consumes `python/requirements/ml/requirements_upstream.txt`?

**Comment by Yard1** [comment]
Our ML docker image. Not sure if there is anything else.

**Comment by cadedaniel** [comment]
> Our ML docker image. Not sure if there is anything else. @amogkam, do you know of anything else?

Does this mean that any product testing on the ML image should restart?

**Comment by Yard1** [comment]
I think we'd just need to rerun xgboost-ray and lightgbm-ray tests. Nothing else should be affected.

**Comment by cadedaniel** [comment]
> I think we'd just need to rerun xgboost-ray and lightgbm-ray tests. Nothing else should be affected.

OK. Can I create a release-blocking issue and assign to you for this? Don't want to lose track

**Comment by cadedaniel** [comment]
Created https://github.com/ray-project/ray/issues/32549. OK, this LGTM, feel free to merge once tests look good!

**Review by cadedaniel — APPROVED** [review]

File: doc/source/ray-air/examples/analyze_tuning_results.ipynb
Comment (angelinalg) [inline doc/source/ray-air/examples/analyze_tuning_results.ipynb:174]:
```suggestion
    "Note that we pass in a `scaling_config` to configure the distributed training behavior of each individual XGBoost training job. We want to distribute training across 2 workers. We also keep some CPU resources free for Ray Data operations.\n",
```

Comment (cadedaniel) [reply doc/source/ray-air/examples/analyze_tuning_results.ipynb]: this suggestion should go into master as well! @angelinalg

**Pull request merged**
